packages/aws-cdk-lib/region-info/build-tools/generate-static-data.ts

```diff
@@ -56,6 +56,7 @@ export async function main(): Promise<void> {
     registerFact(region, 'DOMAIN_SUFFIX', domainSuffix);
 
     registerFact(region, 'CDK_METADATA_RESOURCE_AVAILABLE', AWS_CDK_METADATA.has(region) ? 'YES' : 'NO');
+    registerFact(region, 'ECR_PUBLIC_AVAILABLE', partition === 'aws' ? 'YES' : 'NO');
```

Comment:
Did you consider using the `AWS::Partition` pseudo param in conjunction with `Fn::If` to make this more environment agnostic? its a bummer the user must specify a `region` in the stack in order to get this functionality.